pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -722,8 +732,41 @@ func setUpBuildWaiter(container corev1.Container, waitFile string) corev1.Contai
 
 }
 
-func (b *Build) useStandardContainers(buildWaiterImage string, pod *corev1.Pod) *corev1.Pod {
+func (b *Build) useImageExtensions(pod *corev1.Pod) {
+	lifecycleExperimentalEnvVar := corev1.EnvVar{Name: "CNB_EXPERIMENTAL_MODE", Value: "warn"}
+
+	kanikoVolume := corev1.Volume{
+		Name: "kaniko",
+		VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		},
+	}
+	pod.Spec.Volumes = append(pod.Spec.Volumes, kanikoVolume)
+	kanikoMount := corev1.VolumeMount{
+		Name:      "kaniko",
+		MountPath: "/kaniko",
+	}
+
+	for idx, container := range pod.Spec.InitContainers {
+		container.Env = append(container.Env, lifecycleExperimentalEnvVar)
+		switch container.Name {
+		case RestoreContainerName:
+			container.VolumeMounts = append(container.VolumeMounts, kanikoMount)
+			container.Args = append(container.Args, fmt.Sprintf("-build-image=%s", b.Spec.Builder.Image))
+		case BuildContainerName:
+			container.Name = ExtendContainerName
+			container.Command = []string{"/cnb/lifecycle/extender"}
+			container.VolumeMounts = append(container.VolumeMounts, kanikoMount)
+			container.SecurityContext.RunAsUser = intPointer(0)
+			container.SecurityContext.RunAsGroup = intPointer(0)
+			container.SecurityContext.RunAsNonRoot = boolPointer(false)
+			container.SecurityContext.Capabilities = &corev1.Capabilities{Add: []corev1.Capability{"SETGID", "SETUID"}}
```

Comment:
Discussed in 10/31 Working Group, run as root is unnecessary if all we're doing is switching the runtime base. We should make this configurable by the end user (and default to the most secure setting).